### PR TITLE
[Python] Additional properties with array types

### PR DESCRIPTION
**Pull request by jirikuncar**
Describes issue with `additionalProperties` with `array` types.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: samples/openapi3/client/petstore/python-experimental/petstore_api/model/additional_properties_with_array_of_enums.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          as additional properties values.
+    """
+
+    allowed_values = {
+    }
+
+    validations = {
+    }
+
+    additional_properties_type = ([enum_class.EnumClass],)  # noqa: E501
```
Comment (jirikuncar) [inline samples/openapi3/client/petstore/python-experimental/petstore_api/model/additional_properties_with_array_of_enums.py:68]: ❗ missing import

File: samples/openapi3/client/petstore/python-experimental/petstore_api/api/fake_api.py [hunk 2]
```diff
@@ -36,6 +36,8 @@
 )
 from petstore_api.model import health_check_result
 from petstore_api.model import outer_composite
+from petstore_api.model import additional_properties_with_array_of_enums
+from petstore_api.model import additional_properties_with_named_array_of_enums
```
Comment (jirikuncar) [inline samples/openapi3/client/petstore/python-experimental/petstore_api/api/fake_api.py:40]: ❗ model is not generated

File: samples/openapi3/client/petstore/python-experimental/petstore_api/models/__init__.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 # sys.setrecursionlimit(n)
 
 from petstore_api.model.additional_properties_class import AdditionalPropertiesClass
+from petstore_api.model.additional_properties_with_array_of_enums import AdditionalPropertiesWithArrayOfEnums
```
Comment (jirikuncar) [inline samples/openapi3/client/petstore/python-experimental/petstore_api/models/__init__.py:15]: ❗ missing `additional_properties_with_named_array_of_enums`

File: samples/openapi3/client/petstore/python-experimental/petstore_api/model/additional_properties_with_array_of_enums.py (hunk 1 above)
Comment (jirikuncar) [reply samples/openapi3/client/petstore/python-experimental/petstore_api/model/additional_properties_with_array_of_enums.py:73]: solved https://github.com/OpenAPITools/openapi-generator/pull/6594/files#diff-72e44f37b1cd9b6c6676fba9300b85f2R37

**Comment by jirikuncar** [comment]
Can you please review this fix @taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @Jyhess (2019/01) @arun-nalla (2019/11) @spacether (2019/11)?

File: samples/openapi3/client/petstore/python-experimental/docs/FakeApi.md [hunk 4]
```diff
@@ -332,6 +333,64 @@ No authorization required
...
+
+# Enter a context with an instance of the API client
+with petstore_api.ApiClient() as api_client:
+    # Create an instance of the API class
+    api_instance = fake_api.FakeApi(api_client)
+    
+    # example, this endpoint has no required or optional parameters
+    try:
+        # Additional Properties with Array of Enums
+        api_response = api_instance.get_additional_properties_with_array_of_enums()
```
Comment (spacether) [inline samples/openapi3/client/petstore/python-experimental/docs/FakeApi.md:365]: @jirikuncar why doesn't the documentation example pass in an instance of additional_properties_with_array_of_enums.AdditionalPropertiesWithArrayOfEnums?

Comment (spacether) [reply samples/openapi3/client/petstore/python-experimental/docs/FakeApi.md:365]: Never mind, the endpoint responds with that data type, it doesn't accept it

File: samples/openapi3/client/petstore/java/jersey2-java8/src/main/java/org/openapitools/client/api/FakeApi.java [hunk 5]
```diff
@@ -362,6 +362,66 @@ public String fakeOuterStringSerialize(String body) throws ApiException {
...
+   * 
+   * @return Map&lt;String, List&lt;EnumClass&gt;&gt;
+   * @throws ApiException if fails to make API call
+   * @http.response.details
+     <table summary="Response Details" border="1">
+       <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
+       <tr><td> 200 </td><td> Got object with additional properties with array of enums </td><td>  -  </td></tr>
+     </table>
+   */
+  public Map<String, List<EnumClass>> getAdditionalPropertiesWithArrayOfEnums() throws ApiException {
```
Comment (jirikuncar) [inline samples/openapi3/client/petstore/java/jersey2-java8/src/main/java/org/openapitools/client/api/FakeApi.java:376]: It looks like Java is suffering the same issue with missing import and not using named class for mapping.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

**Comment by jirikuncar** [comment]
@spacether I would prefer to have the Java fix in separate PR.

**Comment by spacether** [comment]
> @spacether I would prefer to have the Java fix in separate PR.

That sounds great. Can you extract the Java update into a separate PR and we can move forward with merging this fix in for python-experimental here?

**Comment by jirikuncar** [comment]
@spacether I have changed only python related files. Why is the Java generated code changed?

**Comment by spacether** [comment]
> @spacether I have changed only python related files. Why is the Java generated code changed?

Hmm it looks like it is because the yaml file is also used by jersey-2 :(
https://github.com/OpenAPITools/openapi-generator/blob/master/bin/configs/java-jersey2-8.yaml

**Comment by jirikuncar** [comment]
@spacether @jimschubert I have created a separate specification for Java.

**Comment by jirikuncar** [comment]
@spacether `-label:"Client: Java"`

**Pull request merged**